pkg/api-server/inspect_endpoints_test.go

```diff
@@ -191,20 +213,409 @@ var _ = Describe("Inspect WS", func() {
 				&core_mesh.TimeoutResource{
 					Meta: &test_model.ResourceMeta{Name: "t-1", Mesh: "default"},
 					Spec: &mesh_proto.Timeout{
-						Sources:      serviceSelector("*", ""),
-						Destinations: serviceSelector("redis", ""),
-						Conf:         samples.Timeout.Conf,
+						Sources: anyService(),
+						Destinations: selectors{
+							serviceSelector("redis", ""),
+						},
+						Conf: samples.Timeout.Conf,
 					},
 				},
 				&core_mesh.HealthCheckResource{
 					Meta: &test_model.ResourceMeta{Name: "hc-1", Mesh: "default"},
 					Spec: &mesh_proto.HealthCheck{
-						Sources:      serviceSelector("backend", ""),
-						Destinations: serviceSelector("*", ""),
+						Sources: selectors{
+							serviceSelector("backend", ""),
+						},
+						Destinations: anyService(),
 						Conf:         samples.HealthCheck.Conf,
 					},
 				},
 			},
 		}),
+		Entry("inspect traffic permission", testCase{
+			path:       "/meshes/default/traffic-permissions/tp-1/dataplanes",
+			goldenFile: "inspect_traffic-permission.json",
+			resources: []core_model.Resource{
+				newMesh("default"),
+				&core_mesh.TrafficPermissionResource{
+					Meta: &test_model.ResourceMeta{Name: "tp-1", Mesh: "default"},
+					Spec: &mesh_proto.TrafficPermission{
+						Sources: anyService(),
+						Destinations: selectors{
+							serviceSelector("backend", "http"),
+							serviceSelector("redis", "http"),
+							serviceSelector("gateway", "http"),
+						},
+					},
+				},
+				newDataplane().
+					meta("backend-1", "default").
+					inbound("backend", "192.168.0.1", 80, 81).
+					outbound("redis", "192.168.0.2", 8080).
+					outbound("gateway", "192.168.0.3", 8080).
+					build(),
```

Comment:
I'm building the same thing on the side :) 

We need resource builders with sample data. For many of tests we don't even care what is the dataplane, yet we still rebuild it.

Reply:
Yeah, that would be awesome to have such builders. Do you need help with the work you're doing?

Reply:
not really, I just need to validate my idea over a bunch of tests